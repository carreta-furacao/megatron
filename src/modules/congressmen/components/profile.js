import React from 'react'
import PropTypes from 'prop-types'
import { Text, View, Image } from 'react-native'

import { profileStyle } from './styles/profile.style'

export const Profile = (props) => {
  const { urlFoto, nome } = props.congressman

  return (
    <View style={profileStyle.container}>
      <Image
        style={profileStyle.picture}
        source={{ uri: urlFoto }}
      />
      <Text style={profileStyle.text}>{nome}</Text>
    </View>
  )
}

Profile.propTypes = {
  congressman: PropTypes.shape({
    nome: PropTypes.string.isRequired,
    urlFoto: PropTypes.string.isRequired
  }).isRequired
}
