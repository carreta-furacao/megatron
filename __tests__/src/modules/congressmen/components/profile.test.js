import React from 'react'
import { shallow } from 'enzyme'

import { Profile } from '../../../../../src/modules/congressmen/components/profile'

const props = {
  congressman: {
    id: 178957,
    uri: 'https://dadosabertos.camara.leg.br/api/v2/deputados/178957',
    nome: 'ABEL MESQUITA JR.',
    siglaPartido: 'DEM',
    uriPartido: 'https://dadosabertos.camara.leg.br/api/v2/partidos/36769',
    siglaUf: 'RR',
    idLegislatura: 55,
    urlFoto: 'http://www.camara.leg.br/internet/deputado/bandep/178957.jpg'
  }
}

describe('<Profile />', () => {
  it('should have a snapshot', () => {
    const wrapper = shallow(<Profile {...props} />)
    expect(wrapper).toMatchSnapshot()
  })
})
