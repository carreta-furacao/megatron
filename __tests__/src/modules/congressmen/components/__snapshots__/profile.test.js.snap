// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> should have a snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "margin": 20,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "http://www.camara.leg.br/internet/deputado/bandep/178957.jpg",
      }
    }
    style={
      Object {
        "height": 100,
        "width": 100,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 20,
        "fontWeight": "bold",
      }
    }
  >
    ABEL MESQUITA JR.
  </Text>
</View>
`;
